blockchain/chain.go

```diff
@@ -1310,8 +1310,21 @@ func (b *BlockChain) reorganizeChain(target *blockNode) error {
 	// actually is here versus using the one from above since it might not match
 	// reality if there were errors while reorganizing.
 	newTip := b.bestChain.Tip()
+	wasLatched := b.isCurrentLatch
 	b.maybeUpdateIsCurrent(newTip)
 
+	// If the chain just latched to current, force the UTXO cache to flush to the
+	// database.  This ensures that the full UTXO set is always flushed to the
+	// database when the chain is current, which allows for fetching up-to-date
```

Comment:
It's certainly not a big deal as I can figure out what it means regardless, but I do think the current verbiage of "when the chain is current" is easy to misconstrue as meaning it keeps it up to date (which actually does happen elsewhere though), where this really only applies when the chain initially becomes current.  Maybe the following small tweak might  help avoid any potential ambiguity?

```suggestion
	// database when the chain becomes current, which allows for fetching up-to-date
```

Reply:
Ya, I actually did mean that it keeps it up to date as the current tip changes so long as it is current.  But like you mentioned, that happens elsewhere, and this logic only applies to when it _becomes_ current, so I agree that your suggestion might help avoid ambiguity.  I went ahead and updated that.